src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -53,7 +55,26 @@ public void onCompletion(RecordMetadata md, Exception e) {
 	}
 }
 
-public class MaxwellKafkaProducer extends AbstractAsyncProducer {
+
+public class MaxwellKafkaProducer extends AbstractProducer {
+	private final LinkedBlockingQueue<RowMap> queue;
+	private final MaxwellKafkaProducerWorker worker;
+
+	public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties, String kafkaTopic) {
+		super(context);
+		this.queue = new LinkedBlockingQueue<>(100);
```

Comment:
Just 🤔  if it's of fixed max size of 100 and if we are sure the queue is mostly blocked on the producer thread(most of the times queue being full), may be `ArrayBlockingQueue` has better performance wrt to upfront memory allocation and no node creation(put) and deletion(dequeu) overhead and GC